mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -32,15 +32,20 @@ import org.apache.spark.sql.types.{ArrayType, StructType}
  * :: Experimental ::
  * Maps a sequence of terms to their term frequencies using the hashing trick.
  */
+@Since("1.2.0")
 @Experimental
-class HashingTF(override val uid: String) extends Transformer with HasInputCol with HasOutputCol {
+class HashingTF @Since("1.2.0") (@Since("1.4.0") override val uid: String) extends Transformer
```

Comment:
The constructor with `uid` should be since `1.4.0`. The default constructor below is since `1.2.0`. Please check others as well.